core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2785,41 +2785,69 @@ public static double power(BigDecimal b0, BigDecimal b1) {
     return Math.pow(b0.doubleValue(), b1.doubleValue());
   }
 
-
   // LN, LOG, LOG10, LOG2
 
-  /** SQL {@code LOG(number, number2)} function applied to double values. */
-  public static @Nullable Double log(double number, double number2, int nullFlag) {
-    if (nullFlag == 1 && number <= 0) {
+  /**
+   * SQL {@code LOG(number, base)} function applied to double values.
+   *
+   * @param dialectflag This parameter is used to specify the SQL dialect for the LOG function.
+   * 0 represents MySQL dialect, 1 represents BigQuery dialect, and 2 represents PostgreSQL dialect.
+   */
+  public static @Nullable Double log(double number, double base, int dialectflag) {
+    if (dialectflag == 0 && number <= 0) {
```

Comment:
you only compare dialectflag with 0.
So this flag should be named something like "nonPositiveIsNull".
The JavaDoc should say: "if true return null for non-positive values"

Reply:
yes, you are right

Reply:
done, thank you